src/commons_codec/model.py

```diff
@@ -11,28 +12,25 @@
 else:
     from backports.strenum import StrEnum  # pragma: no cover
 
+try:
+    from sqlalchemy_cratedb.support import quote_relation_name
+except ImportError:  # pragma: no cover
+    quote_relation_name = None
+
 
 @define(frozen=True)
 class TableAddress:
     schema: str
     table: str
 
-    @property
+    @cached_property
     def fqn(self):
         if not self.schema:
             raise ValueError("Unable to compute a full-qualified table name without schema name")
-        return f"{self.quote_identifier(self.schema)}.{self.quote_identifier(self.table)}"
-
-    @staticmethod
-    def quote_identifier(name: str) -> str:
-        """
-        Poor man's table quoting.
-
-        TODO: Better use or vendorize canonical table quoting function from CrateDB Toolkit, when applicable.
-        """
-        if name and '"' not in name:
-            name = f'"{name}"'
-        return name
+        identifier = f"{self.schema}.{self.table}"
```

Comment:
As schema and table are already separated here, quoting them dedicated would be much safer.

Reply:
Our `quote_relation_name` function currently only accepts a single identifier. Maybe add another one, which accepts the fragments separately?

Reply:
Normally, a quoting function is called per identifier, so for both, schema and name separately. With `quote_relation_name` we already moved into the direction of supporting a full relation name FQN, that's why we named it like that if I remember correctly. We did this afaik as at some usages, schema and name weren't available separately. Yeah maybe leave it like that, just keep in mind that quoting single identifiers is less error prone than quoting an already FQN relation name.

Reply:
Yeah, I had it in mind. In this case, I was just trying to assemble the bricks we have to solve the problem at hand. Every suggestion is super valuable. If you think it is acceptable for now, I will leave it for this iteration, and then, on another one, improve that at https://github.com/crate/sqlalchemy-cratedb/pull/155 going forward. Thanks!